core/src/main.cpp

```diff
@@ -9,6 +9,9 @@
 #ifdef USE_MPI
 #include <mpi.h>
 #endif
+#ifdef USE_OASIS
```

Comment:
In the CMakeLists.txt you need to add close to the end
```
if (ENABLE_OASIS)
   target_include_directories(nextsim PRIVATE ${OASIS_INCLUDES})
endif ()
```

Reply:
It already says
```
target_include_directories(nextsimlib PRIVATE ${OASIS_INCLUDES})
```
at line 159 (within an ``if (ENABLE_OASIS) ... endif ()`` clause). Is that not working?

NB the directory here is ``nextsimlib``, not ``nextsim``, as in your example.

Reply:
When I compiled on my machine, line 159 lead ok to the compilation of the shared lib nextsim.  But `main.cpp` (nor any of the test mains) is not compiled with the lib.  
When compiling the `nextsim` executable target, `main.cpp` is compiled just before being linked with the libs. And `main.cpp` or all the mains that include `ModelMetadata.hpp` now include `oasis_c.h` and do not find it. Adding these lines I got to the end of the compilation but it's "trial and error cmake". Please find a better solution, but I can confirm that something more than line 159 is needed. Also, I don't know at all what `PRIVATE` stands for. There is maybe a different keyword for making the include property apply to depending mains.

Reply:
That's strange because the whole point of the library was to compile it once and then link the main and tests to it. Lines 190 and 191 should take care of that by telling cmake that the main executable requires ``nextsimlib``. The library is also included in the ``target_link_libraries`` command in the ``CMakeLists.txt`` files in the ``test`` subdirectories.

Can you open a separate issue (or use the discussions board: https://github.com/nextsimhub/nextsimdg/discussions), including all the output etc?

Reply:
Before opening an issue (I think a discussion would be better), I just point out that the in the `core/test/CMakeLists.txt` and `physics/test/CMakeLists.txt` there are several `target_include_directories` for the different executables. That's where I draw inspiration. Have a look at time. I guess that the `target_link_libraries` is ok for finding objects at link time, but the `target_include_directories` is needed at compile time.

Reply:
I gave a try with `PUBLIC` instead of `PRIVATE` in the `target_include_directory` for `nextsimlib` and now it works for the mains as well. Are you happy with it (and will change it) or do I open the discussion for something explaining in detail how the properties inheritance works in cmake ?

Reply:
@Thanduriel explained to me why ``PUBLIC`` is the right choice, so I'll change it. It's a bit strange that it works on my setup, but it doesn't matter.